### PR TITLE
Daksaray/8 symbol integration

### PR DESCRIPTION
**Pull request by daksaray**
Added .cc and .h files for number, index, object_color, and modified CMake file accordingly.

### CODE REVIEW

**Comment by daksaray** [comment]
The second commit is based on the changes in number and index codes after Tom's updates. 

object_color codes have not been updated based on the new format yet (they were only moved from h2sl_nsf_nri_mvli to h2sl).

File: src/dcg/dcg.cc [hunk 1]
```diff
@@ -117,6 +126,75 @@ fill_search_spaces( const World* world ){
...
+  /*vector< std::string > object;
+  constraints.push_back( "na" );
+  constraints.push_back( "block" );
+  constraints.push_back( "table" );
+  constraints.push_back( "robot" );*/
+  
+    
+  /*Abstract symbols need to be initialized seperately in inference */
+  vector< std::string > container;
+  constraints.push_back( "group" );
```
Comment (tmhoward) [inline src/dcg/dcg.cc]: switch "constraints" to "container"

Comment (tmhoward) [inline src/dcg/dcg.cc]: "group" can become "na" (need to check on corpus as well)

File: src/dcg/dcg.cc [hunk 2]
```diff
@@ -117,6 +126,75 @@ fill_search_spaces( const World* world ){
...
+  vector< std::string > index;
+  constraints.push_back( "first" );
+  constraints.push_back( "second" );
+  constraints.push_back( "third" );
+  constraints.push_back( "fourth" );
+  constraints.push_back( "fifth" );
+  constraints.push_back( "na" );
+    
+  vector< std::string > number;
+  constraints.push_back( "na" );
```
Comment (tmhoward) [inline src/dcg/dcg.cc]: switch "constraints" to "number"

File: src/dcg/dcg.cc [hunk 3]
```diff
@@ -117,6 +126,75 @@ fill_search_spaces( const World* world ){
   vector< std::string > constraints;
   constraints.push_back( "inside" );
   constraints.push_back( "outside" );
+    
+  vector< std::string > index;
+  constraints.push_back( "first" );
```
Comment (tmhoward) [inline src/dcg/dcg.cc]: switch "constraints" to "index"

**Comment by arkinj** [comment]
Is this pull request no longer desired because we now have the ADCG branch?

**Comment by daksaray** [comment]
I believe so. Because Rohan has created the adcg branch based on daksaray/8_symbol_integration.

**Pull request closed without merging**